install_config/router/default_haproxy_router.adoc

```diff
@@ -1487,6 +1487,94 @@ add routes across the namespaces.
 ====
 
 
+[[using-the-dynamic-configuration-manager]]
+== Using the Dynamic Configuration Manager
+
+The HAProxy router supports a dynamic configuration manager to bring certain
+types of routes online without requiring any haproxy reload downtime.
+The dynamic configuration manager support can be enabled by setting
+the `ROUTER_HAPROXY_CONFIG_MANAGER` environment variable to `true`.
+When the dynamic configuration manager is enabled, it will handle any route and
+endpoint lifecyle events such as route and endpoint `addition|deletion|update`.
+In the previous releases if any of these events caused the configuration to
+change, it resulted in the router requiring a full configuration re-write and a
+haproxy process reload.
+
+The dynamic configuration internally uses the haproxy socket and configuration
+API with a pool of pre-allocated routes and backend servers to dynamically
+achieve the same results without requiring a haproxy process reload.
+In the event the dynamic configuration manager is unable to configure haproxy
+dynamically, it forces a configuration re-write and a haproxy process reload.
+This can occur in cases like if a newly added route contains custom annotations
+that would cause a drift from the standard configuration (custom timeouts) or
+if the route requires custom TLS configuration.
+The pre-allocated pool of routes is created using route blueprints - the default
+set of blueprints supports unsecured routes, edge secured routes (without any
+custom TLS configuration) and passthrough routes.
+`re-encypt` routes require custom TLS configuration information and so are not
+supported out of the box. However that said, the blueprints that the dynamic
+configuration manager uses can be extended by setting the `ROUTER_BLUEPRINT_ROUTE_NAMESPACE`
+and optionally the `ROUTER_BLUEPRINT_ROUTE_LABELS` environment variable values.
```

Comment:
Can users configure multiple `ROUTE_NAMESPACE` and `ROUTE_LABELS` for custom blueprints?

Reply:
It depends on permissions to that project/namespace `ROUTE_NAMESPACE`. 
The general use case would be no - as the admin owns the `ROUTE_NAMESPACE` so only the admin can add a `LABEL` to those blueprint routes. But an admin can always grant access to another user to modify that namespace and/or the routes.